test/serializer/basic/LimitedStack.js

```diff
@@ -0,0 +1,18 @@
+// limit stack depth: 100
+// exceeds stack depth limit
+
+(function() {
+  let immediate = function(future) {
+    if (Date.now() > future) {
+      return "exit value";
+    }
+
+    return immediate(future);
+  };
+
+  let n = immediate(Date.now() - 1);
```

Comment:
You pass in a value from the past, not future. In the future, when Prepack might get smarter about value domains, Prepack might figure this out. Either pass in a large value that's actually guaranteed to be in the future, or change the immediate function to do something with past values to recurse.